appIcons.js

```diff
@@ -500,7 +510,7 @@ const MyAppIconMenu = new Lang.Class({
             this._quitfromDashMenuItem = this._appendMenuItem(quitFromDashMenuText);
             this._quitfromDashMenuItem.connect('activate', Lang.bind(this, function() {
                 let app = this._source.app;
-                let windows = app.get_windows();
```

Comment:
This change modifiess the "Quit" button, to only quit windows in the current workspace